File: QE-Selenium-Java/Activity16.java
package demos;

import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.firefox.FirefoxDriver;
import org.openqa.selenium.support.ui.Select;

import java.util.List;

public class Activity16 {

    public static void main(String[] args) {

        WebDriver driver = new FirefoxDriver();

        try {
            driver.get("https://training-support.net/webelements/selects");
            System.out.println("Page Title: " + driver.getTitle());

            WebElement dropdown = driver.findElement(
                    By.xpath("//select"));

            Select select = new Select(dropdown);

            select.selectByVisibleText("Two");

            select.selectByIndex(3); // Four

            select.selectByValue("five");

            System.out.println("Dropdown options:");
            List<WebElement> options = select.getOptions();

            for (WebElement option : options) {
                System.out.println(option.getText());
            }

        } finally {
            driver.quit();
        }
    }
}

File: QE-Selenium-Python/Activity16.py
from selenium import webdriver
from selenium.webdriver.common.by import By
from selenium.webdriver.support.ui import Select
import time

driver = webdriver.Firefox()

try:
    driver.get("https://training-support.net/webelements/selects")
    print("Page Title:", driver.title)

    dropdown = driver.find_element(By.XPATH, "//select")

    select = Select(dropdown)

    select.select_by_visible_text("Two")

    select.select_by_index(3)   # Four

    select.select_by_value("five")

    print("Dropdown options:")
    options = select.options

    for option in options:
        print(option.text)

    time.sleep(3)

finally:
    driver.quit()
